### DISCUSSION TITLE
7D2D dedicated server on ubuntu

### DISCUSSION THREAD

**Discussion by ehkart** [GameServerManagers/LinuxGSM#4630] [Q&A]
I run a private 7D2D dedicated server for myself, family and friends on an Ubuntu 22.04 machine. I use the LinuxGSM tool for any game servers out of convenience. I have had no issues with any other game server; only with 7D2D. When I had first set up my server I tried using the RWG settings in the sdtdserver.xml and encountered an issue; a player connects and begins to move around etc, suddenly the server receives a shutdown command via telnet and the player is booted. I attempted to use different settings, read through countless documents on ensuring everything was dialed in however I kept encountering the same issue. I abandon the RWG settings for the default Nav map suspecting that there was some resource issue regarding map size, generation etc. The map ran perfectly fine for weeks, reliably and without issue until the server was rebooted after an update and suddenly I am facing the exact same issue again. I suspect now it has something to do with the monitor script possibly? I have 3 cron jobs to check for updates and restart the server as necessary but they're verbatim from the documentation. I could use some help to determine what the issue is so I can fix it because my closest solution is abandoning the game entirely.

**Comment by Bozz022187** [comment]
I've been having issues with 7d2d also, i don't know your specific answer, my problem is the start parameters. I've done everything by the box on linuxGSM and can have my map open, but i can't make it public, I'm assuming since i have opened all ports, updated the xml files (just like my server name and such, basic stuff) i can connect to it through 127.0.0.1 but its not discoverable in the game from the servers. This is the first game I'm serving up, but would love to migrate my discords saved game files over to my dedicated server. I've gotten to the finish line and fell. Any advice?

**Reply by Burtyhaxx** [reply]
hey im new to all of this but i wasnt able to access my server, i had used port triggering in my router but appaerntly i had to do it on a virtual server on my router (this wasnt a problem on my old router which had the standard port forwarding) but i also opened the ports in my ubuntu server

apt install ufw
sudo ufw enable
sudo ufw allow add your port number here/tcp
sudo ufw reload

what i read is if you need both tcp and udp, you will have to enter both commands seperatly
 I hope this helps but again, im really new to linux and servers
